src/pages/Loading.tsx

```diff
@@ -14,39 +14,71 @@ const Loading: React.FC<LoadingProps> = ({ condition, error }) => {
   const [isLoading, setIsLoading] = useState(true);
   const [showMain, setShowMain] = useState(false);
   const [isDataFetched, setIsDataFetched] = useState(false);
+  const [progress, setProgress] = useState(0);
+  const [animatedProgress, setAnimatedProgress] = useState(0);
+  const [currentStage, setCurrentStage] = useState("분석 준비중입니다");
 
   useEffect(() => {
     if (error) {
       setIsLoading(false);
     } else if (condition) {
       setIsLoading(false);
-
       setTimeout(() => {
         window.close();
       }, 3000);
     }
   }, [condition, error]);
 
   useEffect(() => {
-    const messageListener = (message: { action: string }) => {
-      if (message.action === "dataFetched") {
-        if (message) {
-          setIsDataFetched(true);
-          setIsLoading(false);
-        } else {
-          console.error("Error fetching data:", message);
-          setIsLoading(false);
+    const eventSource = new EventSource(
+      import.meta.env.VITE_SERVER_URL + "/progress",
+    );
+
+    eventSource.onmessage = (event) => {
+      const data = JSON.parse(event.data);
+
+      if (data.progress) {
+        setCurrentStage(data.stage);
+        setProgress(data.progress);
+
+        if (data.progress === 100) {
+          eventSource.close();
         }
       }
     };
 
-    chrome.runtime.onMessage.addListener(messageListener);
+    eventSource.onerror = (error) => {
+      console.error("SSE Error:", error);
+      setIsLoading(false);
+      setIsDataFetched(false);
+    };
 
     return () => {
-      chrome.runtime.onMessage.removeListener(messageListener);
+      eventSource.close();
     };
   }, []);
 
+  useEffect(() => {
+    let interval: number;
+
+    if (animatedProgress < progress) {
+      interval = window.setInterval(() => {
+        setAnimatedProgress((prev) => Math.min(prev + 1, progress));
+      }, 50);
+    } else if (animatedProgress < 100 && progress === 100) {
+      interval = window.setInterval(() => {
+        setAnimatedProgress((prev) => Math.min(prev + 1, 100));
+      }, 50);
+    } else if (animatedProgress === 100) {
+      setIsDataFetched(true);
+      setIsLoading(false);
+    }
+
+    return () => {
+      if (interval) window.clearInterval(interval);
```

Comment:
한줄이라 괜찮긴 하지만 양식 통일을 위해 중괄호 넣는게 좋을거같은데  어떨까요?
```suggestion
      if (interval) {
        window.clearInterval(interval);
    }
```